src/state/persist.rs

```diff
@@ -107,16 +115,24 @@ impl NetworkState {
             OffsetLen,
         >::new(id.origin);
 
+        let restored_config: &<u64 as Archive>::Archived =
+            store.get::<u64>(&config_ident);
         let restored_head: &<Hamt<ContractId, Contract, (), OffsetLen> as Archive>::Archived =
             store.get::<Hamt<ContractId, Contract, (), OffsetLen>>(&head_ident);
         let restored_origin: &<Hamt<ContractId, Contract, (), OffsetLen> as Archive>::Archived =
             store.get::<Hamt<ContractId, Contract, (), OffsetLen>>(&origin_ident);
 
+        let restored_config_hash: u64 =
+            restored_config.deserialize(&mut store.clone()).unwrap();
         let restored_head: Hamt<ContractId, Contract, (), OffsetLen> =
             restored_head.deserialize(&mut store.clone()).unwrap();
         let restored_origin: Hamt<ContractId, Contract, (), OffsetLen> =
             restored_origin.deserialize(&mut store.clone()).unwrap();
 
+        if restored_config_hash != config_hash(self.config) {
+            return Err(VMError::ConfigMismatch);
```

Comment:
As discussed, do we want every config change to enforce incompatibility? Comparing hashes will make every config change a critical change. If we had a version stored rather than hash, we could have more control over what's compatible and what is not.

Reply:
I'll remove this feature. It doesn't belong in this PR and is stalling progress.